src/fuzz_introspector/analyses/sinks_analyser.py

```diff
@@ -0,0 +1,394 @@
+# Copyright 2022 Fuzz Introspector Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Analysis plugin for introspection sinks of interest"""
+
+import logging
+
+from typing import (
+    List,
+    Tuple,
+    Dict
+)
+
+from fuzz_introspector import (
+    analysis,
+    cfg_load,
+    html_helpers,
+    utils
+)
+
+from fuzz_introspector.datatypes import (
+    project_profile,
+    fuzzer_profile,
+    function_profile
+)
+
+logger = logging.getLogger(name=__name__)
+
+# Common sink functions / methods for different language implementation
+SINK_FUNCTION = {
+    'c-cpp': [
+        ('', 'system'),
+        ('', 'execl'),
+        ('', 'execve'),
+        ('', 'wordexp'),
+        ('', 'popen'),
+        ('', 'fdopen')
+    ],
+    'python': [
+        ('', 'exec'),
+        ('', 'eval'),
+        ('subprocess', 'call'),
+        ('subprocess', 'run'),
+        ('subprocess', 'Popen'),
+        ('subprocess', 'check_output'),
+        ('os', 'system'),
+        ('os', 'popen'),
+        ('os', 'spawnlpe'),
+        ('os', 'spawnve'),
+        ('os', 'execl'),
+        ('os', 'execve'),
+        ('asyncio', 'create_subprocess_shell'),
+        ('asyncio', 'create_subprocess_exec'),
+        ('asyncio', 'run'),
+        ('asyncio', 'sleep'),
+        ('logging.config', 'listen'),
+        ('code.InteractiveInterpreter', 'runsource'),
+        ('code.InteractiveInterpreter', 'runcode'),
+        ('code.InteractiveInterpreter', 'write'),
+        ('code.InteractiveConsole', 'push'),
+        ('code.InteractiveConsole', 'interact'),
+        ('code.InteractiveConsole', 'raw_input'),
+        ('code', 'interact'),
+        ('code', 'compile_command')
+    ],
+    'jvm': [
+        ('java.lang.Runtime', 'exec'),
+        ('javax.xml.xpath.XPath', 'compile'),
+        ('javax.xml.xpath.XPath', 'evaluate'),
+        ('java.lang.Thread', 'sleep'),
```

Comment:
We should add some documentation on why each function is relevant, even just a link if it's based on a previous vulnerability or so. Am not sure if it's general knowledge to users why `java.lang.Thread.sleep` is an indication of possible code injection, but to me it's not.

Reply:
@arthurscchan could you please follow up on this?